internal/generator/vector/source/kubernetes_logs.go

```diff
@@ -36,42 +42,94 @@ rotate_wait_ms = 5000
 {{end}}`
 }
 
-func NewKubernetesLogs(id, namespace string) KubernetesLogs {
+func NewKubernetesLogsForOpenShiftLogging(id string) KubernetesLogs {
+
+	excludes := []string{}
+	for _, comp := range []string{constants.LogfilesmetricexporterName, constants.ElasticsearchName, constants.KibanaName} {
+		comp = fmt.Sprintf(`"/var/log/pods/%s_%s-*/*/*.log"`, constants.OpenshiftNS, comp)
+		excludes = append(excludes, comp)
+	}
+	for _, comp := range []string{"loki*", "gateway", "opa"} {
+		comp = fmt.Sprintf(`"/var/log/pods/%s_*/%s/*.log"`, constants.OpenshiftNS, comp)
+		excludes = append(excludes, comp)
+	}
+	for _, comp := range []string{"gz", "tmp"} {
+		comp = fmt.Sprintf(`"/var/log/pods/*/*/*.%s"`, comp)
+		excludes = append(excludes, comp)
+	}
+	return NewKubernetesLogs(id, "", joinContainerPathsForVector(excludes))
+}
+
+// NewKubernetesLogs element which always excludes temp and gzip files
+func NewKubernetesLogs(id, includes, excludes string) KubernetesLogs {
 	return KubernetesLogs{
 		ComponentID:  id,
 		Desc:         "Logs from containers (including openshift containers)",
-		ExcludePaths: ExcludeContainerPaths(namespace),
+		IncludePaths: includes,
```

Comment:
How does this get populated?  Is this a stub for future efforts?

Reply:
It's here for future when we implement the input APIs and allow users to include or exclude namespaces or containers